### PR TITLE
Upgrade main to `go1.18.1`

### PR DESCRIPTION
**Pull request by frouioui**
## Description
This Pull Request upgrades the golang version used on `main` to `go1.18.1`. The Go team has recently released a new patch release for `go1.18` that contains vulnerability fixes, that impact Vitess.

Snippet from the [Go release history website](https://go.dev/doc/devel/release#go1.18.minor):
> go1.18.1 (released 2022-04-12) includes security fixes to the crypto/elliptic, crypto/x509, and encoding/pem packages, as well as bug fixes to the compiler, linker, runtime, the go command, vet, and the bytes, crypto/x509, and go/types packages. See the [Go 1.18.1 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.18.1+label%3ACherryPickApproved) on our issue tracker for details.

Note that the minimum required Go version is `1.18.1`, if this requirement is not met, the following error will appear when building Vitess:
```
Version 1.18.1+ required. See https://vitess.io/contributing/build-from-source for install instructions.
```

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/servenv/buildinfo_test.go [hunk 1]
```diff
@@ -33,17 +33,17 @@ func TestVersionString(t *testing.T) {
 		buildTimePretty: "time is now",
 		buildGitRev:     "d54b87ca0be09b678bb4490060e8f23f890ddb92",
 		buildGitBranch:  "gitBranch",
-		goVersion:       "1.18",
+		goVersion:       "1.18.1",
 		goOS:            "amiga",
```
Comment (systay) [inline go/vt/servenv/buildinfo_test.go:37]: `goOS: "amiga"` 😆

**Review by systay — APPROVED** [review]

File: .github/docker/cluster_test_12/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
 
-ARG bootstrap_version=5
+ARG bootstrap_version=5.1
```
Comment (deepthi) [inline .github/docker/cluster_test_12/Dockerfile:3]: on the main branch, it is appropriate to bump this number up to 6. We just can't do that on release branches.

**Review by deepthi — CHANGES_REQUESTED** [review]
Let's change bootstrap_version to 6, and upload new bootstrap images.
I can re-review and approve once that is done.

File: .github/docker/cluster_test_12/Dockerfile (hunk 2 above)
Comment (frouioui) [reply .github/docker/cluster_test_12/Dockerfile:3]: Makes sense!

**Review by deepthi — APPROVED** [review]

**Pull request merged**